R/partial-factor.R

```diff
@@ -0,0 +1,102 @@
+#' Partially specify a factor
+#'
+#' This special class can be passed as a `ptype` in order to specify that the
+#' result should be a factor with at least some minimum set of levels.
+#'
+#' @inheritParams new_factor
+#' @export
+#' @examples
+#' # Assert that `x` is a factor with any levels
```

Comment:
```suggestion
#' # Assert that `x` is a factor
```

Maybe also discuss what `vec_assert(factor("x"), factor())`  does?